### DISCUSSION TITLE
Cypress keeps disconnecting the websocket connection

### DISCUSSION THREAD

**Discussion by snehaviswanathan** [cypress-io/cypress#16338] [Questions and Help]
Hello All

I am new to cypress so please pardon me if I ask any silly question.
My application streams data, so opens up a websocket connection and then streams the data continuously & closes it when it's completed. The streaming starts when microphone has access. This works completely fine on actual browser when trying it out manually but when I run my tests which is entirely UI driven through cypress on FF & Chrome, it disconnects the stream as soon as it opens. The application tries to reconnect to the websocket, but it keeps disconnecting. When I tried with DEBUG the code when the socket gets closed is 1006 with empty reason. Since it doesn't happen on actual browser, is the testing framework causing the disconnect or is there any specific setting which I need to add ? 

I have gone through few questions but they are mainly around mocking the websocket, which is not what I am looking for.

Appreciate any help.

Thanks

**Comment by portyu9** [comment]
Hi @snehaviswanathan 

Cypress supports WebSockets, so there is no setting required to keep the socket open.

In this case the likely difference is the **microphone**, not the WebSocket. Cypress launches Chrome with fake media-device options, so your application may not be receiving the same audio stream as it does in a normal browser.

If you need the real microphone, remove Cypress's fake-media flags:

```js
setupNodeEvents(on) {
  on('before:browser:launch', (browser, launchOptions) => {
    if (browser.family === 'chromium') {
      launchOptions.args = launchOptions.args.filter(
        (arg) =>
          arg !== '--use-fake-ui-for-media-stream' &&
          arg !== '--use-fake-device-for-media-stream'
      )
    }

    return launchOptions
  })
}
```

For an automated/CI test, an even better approach is to provide a deterministic audio fixture with Chrome's `--use-file-for-fake-audio-capture` option.

The `1006` code only means the WebSocket closed abnormally; it does not identify Cypress as the cause. If it still disconnects after supplying a valid continuous audio stream, inspect the WebSocket upgrade/server logs for the actual reason.

Please mark this answer as accepted if it helped.
